### DISCUSSION TITLE
Is it possible to use aliases defined in vite.config.ts in tag helpers?

### DISCUSSION THREAD

**Discussion by bgr11n** [ElMassimo/vite_ruby#349] [Q&A] [answered]
Hi

I have following file structure:
```ruby
# app
# |  assets
# |  |  stylesheets
# |  |  |  my_style.css
# |  |  images
# |  |  |  my_image.jpg
# |  javascript
# |  |  entrypoints
# |  views
# |  |  application
# |  |  |  home.html.slim
```
Here is vite.config.ts:
```ts
import { resolve } from 'path';
import { defineConfig } from 'vite';

export default defineConfig({
  resolve: {
    alias: {
      '@assets': resolve('/app/app/assets'),
      '@stylesheets': resolve('/app/app/assets/stylesheets'),
      '@images': resolve('/app/app/assets/images')
    },
  },
})
```

And when in `home.html.slim` I am trying to do:
```slim
img src=vite_asset_path('@images/triptix.jpg')
```
It is not working, server is returning 404 error. Only this way is working:
```slim
img src=vite_asset_path('/app/assets/images/triptix.jpg')

Am I understood correctly that this should not suppose to work or am I doing something wrong?

Thank you!
```

**Comment by ElMassimo** [comment, marked as answer]
Hi Bohdan!

Aliases are only applied to import paths in Vite.

If possible, move any images you'll use in tag helpers to `<sourceCodeDir>/images` which would allow you to use `vite_asset_path('images/triptix.jpg')` (with the default configuration).

---

Otherwise, in order to reference files outside of the source code dir using tag helpers, you'll need to add them as entrypoints.


See [this example](https://vite-ruby.netlify.app/guide/advanced.html#one-entrypoint-per-component), and read the docs for [`additionalEntrypoints`](https://vite-ruby.netlify.app/config/#additionalentrypoints).

**Reply by bgr11n** [reply]
@ElMassimo Got it, thank you!
